Packages/Fluxity/Runtime/Dispatcher.cs

```diff
@@ -31,14 +33,33 @@ public void RegisterEffect<TCommand>(IEffect<TCommand> effect)
         public void Dispatch<TCommand>(TCommand command)
             where TCommand : ICommand
         {
-            OnDispatch?.Invoke(command);
-            if (_isReducing)
-                throw new DispatcherException($"Attempted to dispatch '{command.GetType()}' during an existing {nameof(Dispatch)}.");
+            _commandQueue.Enqueue(command);
+            if (_isDequeing) return;
 
-            _isReducing = true;
-            _store.ProcessCommand(command);
-            _isReducing = false;
+            InnerDispatch(command);
+        }
 
+        private void InnerDispatch<TCommand>(TCommand command) where TCommand : ICommand
+        {
+            _isDequeing = true;
+            var count = 0;
+            while (_commandQueue.Count > 0)
+            {
+                var nextCommand = _commandQueue.Dequeue();
+                ProcessCommand(nextCommand);
+                if (count++ > MaxDequeueDispatches)
+                {
+                    throw new DispatcherException($"Dispatcher is stuck in a loop. Check for circular dependencies in your effects. " +
```

Comment:
I'd say exceeded queue limit rather than in a loop because not guaranteed to be in a loop.